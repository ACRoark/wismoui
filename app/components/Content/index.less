@import (reference) '../../styles/themes/index.less';

@content-cls-prefix: ~'@{dte-prefix}-content';

.@{content-cls-prefix} {
  padding-bottom: @footer-height;

  @media @mobile {
    padding-bottom: @footer-mobile-height;
  }
}
