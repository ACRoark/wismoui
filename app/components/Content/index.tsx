import React, { FC, ReactElement, ReactNode } from 'react';

import './index.less';

interface IContentProps {
  children: ReactNode;
}

const Content: FC<IContentProps> = (props: IContentProps): ReactElement => (
  <div className="dte-wismo-content">{props.children}</div>
);

export default Content;
