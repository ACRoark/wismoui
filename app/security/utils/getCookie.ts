const getCookie = (name: string): string =>
  document.cookie.split('; ').reduce((previousValue: string, currentValue: string): string => {
    const parts = currentValue.split('=');

    return parts[0] === name ? decodeURIComponent(parts[1]) : previousValue;
  }, '');

export default getCookie;
