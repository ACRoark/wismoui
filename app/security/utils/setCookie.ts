interface ICookieOptions {
  days: number;
  path: string;
}

const setCookie = (name: string, value: string, options: ICookieOptions): void => {
  const expires = new Date(Date.now() + options.days * 864e5).toUTCString();

  document.cookie = `${name}=${encodeURIComponent(value)}; expires=${expires}; path=${options.path}`;
};

export default setCookie;
