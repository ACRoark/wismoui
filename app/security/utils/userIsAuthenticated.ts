import useAuth from 'hooks/useAuth';

const userIsAuthenticated = (): boolean => {
  const { isAuthenticated } = useAuth();

  return isAuthenticated;
};

export default userIsAuthenticated;
