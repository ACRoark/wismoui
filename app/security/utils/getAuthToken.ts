import getCookie from './getCookie';

const getAuthToken = (): string => getCookie('token');

export default getAuthToken;
