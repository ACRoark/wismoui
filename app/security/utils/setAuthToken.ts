import setCookie from './setCookie';

const setAuthToken = (value?: string): void => {
  if (value) {
    setCookie('token', `Bearer ${value}`, { days: 1, path: '/' });
  }
};

export default setAuthToken;
