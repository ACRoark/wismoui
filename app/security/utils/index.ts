import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
