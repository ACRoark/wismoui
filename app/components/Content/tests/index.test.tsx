import React from 'react';

import { render } from 'enzyme';
import { createSnapshot } from 'testing/utils';

import Content from '../index';

const Children = (): React.ReactElement => <div>Children</div>;

describe('<Content />', (): void => {
  it('should not log errors in console when rendered', (): void => {
    const spy = jest.spyOn(global.console, 'error');

    render(
      <Content>
        <Children/>
      </Content>
    );

    expect(spy).not.toHaveBeenCalled();
  });

  it('should render correctly when passed valid children', (): void => {
    const tree = createSnapshot(
      <Content>
        <Children/>
      </Content>
    );

    expect(tree).toMatchSnapshot();
  });
});
