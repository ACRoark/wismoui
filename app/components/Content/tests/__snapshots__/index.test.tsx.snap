// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> should render correctly when passed valid children 1`] = `
<div
  className="dte-wismo-content"
>
  <div>
    Children
  </div>
</div>
`;
